pkg/spec/spec.go

```diff
@@ -32,7 +32,8 @@ type Function struct {
 // FunctionSpec contains func specification
 type FunctionSpec struct {
 	Handler  string             `json:"handler"`
-	Function string             `json:"function"`
+	File     string             `json:"file"`
```

Comment:
A single file?
I feel like if we're redesigning this, then we should make it a lot more future-proof.  Perhaps something like a list of enumerated types, where URL, literal, etc are just specific types.  Need to be careful, because we're sort of reimplementing k8s volumes (and so we just use them here or have a clear reason why not).

Reply:
Actually rethinking about this, it will allow us to avoid breaking changes, other clients like the the UI can keep using the field to specify the file content so we don't need to implement Minio support for every client.